polysemy-plugin/src/Polysemy/Plugin.hs

```diff
@@ -61,51 +61,34 @@ module Polysemy.Plugin
   ( plugin
   ) where
 
-import CoreMonad
-import DynFlags
-import GHC (ModuleName, moduleName)
-import Module (mkModuleName, moduleSetElts)
+import GhcPlugins
+
 import Polysemy.Plugin.Fundep
 import Polysemy.Plugin.InlineRecursiveCalls
-
 #if __GLASGOW_HASKELL__ >= 810
 import Polysemy.Plugin.Phases
 #endif
 
-import Plugins (Plugin (..), defaultPlugin)
-#if __GLASGOW_HASKELL__ >= 806
-import Plugins (PluginRecompile(..))
-#endif
-
-
+------------------------------------------------------------------------------
 plugin :: Plugin
 plugin = defaultPlugin
-    { tcPlugin = const $ Just fundepPlugin
-    , installCoreToDos = const installTodos
+    { tcPlugin            = const $ Just fundepPlugin
 #if __GLASGOW_HASKELL__ >= 806
-    , pluginRecompile = const $ pure NoForceRecompile
+    , pluginRecompile     = purePlugin
+    , renamedResultAction = \_ env group ->
+        traverse inlineRecCallsAction (env, group)
+    , installCoreToDos    = const installTodos
```

Comment:
You can still do this on older versions

Reply:
`installCoreToDos`? Is it useful with `renamedResultAction` absent?

Reply:
It's always useful, but only on GHC >= 8.10 --- maybe we could make that more clear here.

Reply:
Should I move it outside and add `-- Passes to get more out of 8.10 specializing`?